{
  "Name": "CurseClient",
  "Process": "Curse.Agent.Host",
  "PID": 2233,
  "Arguments": [
    "/tmp/.mount_CurseFlFVKY1/resources/app.asar.unpacked/plugins/curse/linux/Curse.Agent.Host",
    "--port=62268",
    "--to-port=62288",
    "--pid=1978",
    "--appDataPath=/home/avery/.config/CurseForge",
    "--clientId=Overwolf"
  ],
  "ProcessStart": "2025-06-11T17:53:23.7201214Z",
  "Version": "1.279.0.26201",
  "BuildDate": "2000-01-01T19:33:22Z",
  "Start": "2025-06-11T13:53:25.2934196-04:00",
  "Log": [
{
  "timestamp": "2025-06-11T13:53:25.2735142-04:00",
  "level": "Warn",
  "thread": 1,
  "memory": 0.0,
  "message": "Failed to modified BuiltInConverters: Cannot set initonly static field 'BuiltInConverters' after type 'Newtonsoft.Json.Serialization.DefaultContractResolver' is initialized."
},
{
  "timestamp": "2025-06-11T13:53:25.2853765-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Startup Name: CurseForge, Version: 1.279.0.26201"
},
{
  "timestamp": "2025-06-11T13:53:25.287016-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Running Mods Agent with args: --port=62268,--to-port=62288,--pid=1978,--appDataPath=/home/avery/.config/CurseForge,--clientId=Overwolf"
},
{
  "timestamp": "2025-06-11T13:53:25.2965397-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "start monitor process: 1978"
},
{
  "timestamp": "2025-06-11T13:53:27.3591413-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Unable to load game instances from disk. No file was found at: /home/avery/.config/CurseForge/agent/GameInstances/ModpackGameInstances.json"
},
{
  "timestamp": "2025-06-11T13:53:27.61487-04:00",
  "level": "Info",
  "thread": 1,
  "memory": 0.0,
  "message": "Mods Agent server started on port 62268"
},
{
  "timestamp": "2025-06-11T13:53:27.7106091-04:00",
  "level": "Info",
  "thread": 8,
  "memory": 0.0,
  "message": "Web Socket session connected"
},
{
  "timestamp": "2025-06-11T13:53:28.0682337-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Update simultaneous downloads '1' => '3'"
},
{
  "timestamp": "2025-06-11T13:53:28.078376-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Initializing Curse Mods Agent"
},
{
  "timestamp": "2025-06-11T13:53:28.083131-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "update client Info ''..."
},
{
  "timestamp": "2025-06-11T13:53:28.2089015-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-11T13:53:28.2503621-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "AddonServiceClient - CfCore Url: https://api.curseforge.com"
},
{
  "timestamp": "2025-06-11T13:53:28.4180621-04:00",
  "level": "Warn",
  "thread": 14,
  "memory": 0.0,
  "message": "Service call failed to: legacy/game/timestamp",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.DispatchWorkItem(Object,Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:53:28.4747014-04:00",
  "level": "Error",
  "thread": 14,
  "memory": 0.0,
  "message": "Failed getting games database timestamp from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-11T13:53:28.4763048-04:00",
  "level": "Error",
  "thread": 14,
  "memory": 0.0,
  "message": "Failed getting games database timestamp from server",
  "error": {
    "message": "Failed to get games timestamp from server",
    "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
    "stack": [
      "Curse.TechService.AddonServiceClient+<GetGameDatabaseTimestampAsync>d__31.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.TechService.Games.GameRepository+<GetServiceTimestampAsync>d__14.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.TechService.AddonServiceClient+<GetGameDatabaseTimestampAsync>d__31.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task`1.TrySetResult(TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1,TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult(TResult)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.DispatchWorkItem(Object,Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ]
  }
},
{
  "timestamp": "2025-06-11T13:53:28.4862552-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on GameDatabase -> GetRemoteItems"
},
{
  "timestamp": "2025-06-11T13:53:28.4877692-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Getting Games from server"
},
{
  "timestamp": "2025-06-11T13:53:28.5106905-04:00",
  "level": "Warn",
  "thread": 6,
  "memory": 0.0,
  "message": "Service call failed to: legacy/game?supportsAddons=true",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.DispatchWorkItem(Object,Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:53:28.5133932-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-11T13:53:33.5153881-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on GameDatabase -> GetRemoteItems"
},
{
  "timestamp": "2025-06-11T13:53:33.5155711-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Getting Games from server"
},
{
  "timestamp": "2025-06-11T13:53:33.5319943-04:00",
  "level": "Warn",
  "thread": 8,
  "memory": 0.0,
  "message": "Service call failed to: legacy/game?supportsAddons=true",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishContinuations()",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:53:33.5412557-04:00",
  "level": "Error",
  "thread": 8,
  "memory": 0.0,
  "message": "Failed getting games from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-11T13:53:43.5466263-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games from server",
  "error": {
    "message": "One or more errors occurred. (Failed to get games from server) (Failed to get games from server)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.TechService.Games.GameRepository+<GetRemoteItemsAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishStageThree()",
      "System.Threading.Tasks.Task.FinishStageTwo()",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.Finish(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()",
      "System.Threading.TimerQueueTimer.Fire(Boolean)",
      "System.Threading.TimerQueue.FireNextTimers()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed to get games from server",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient+<GetGamesAsync>d__30.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.TechService.Games.GameRepository+<<GetRemoteItemsAsync>b__16_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:53:43.5626018-04:00",
  "level": "Warn",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting games list from server. Attempting to load from local cache"
},
{
  "timestamp": "2025-06-11T13:53:43.5639399-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Loading games from local cache at /home/avery/.config/CurseForge/agent/gamesCache.json"
},
{
  "timestamp": "2025-06-11T13:53:44.5800365-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Loaded 25 games from the local games cache"
},
{
  "timestamp": "2025-06-11T13:53:44.6012998-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "[GameRepository] Completed download of cache",
  "data": {
    "name": "Games",
    "elapsedTime": "16,097.00"
  }
},
{
  "timestamp": "2025-06-11T13:53:44.6487311-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing local databases..."
},
{
  "timestamp": "2025-06-11T13:53:44.649549-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing game instances."
},
{
  "timestamp": "2025-06-11T13:53:44.6858869-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Found RegisterType on Type: 'Curse.Radium.Addons.AddonGameInstance'"
},
{
  "timestamp": "2025-06-11T13:53:44.6868193-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Found RegisterType on Type: 'Curse.Games.Minecraft.MinecraftGameInstance'"
},
{
  "timestamp": "2025-06-11T13:53:44.6873033-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing game instance of type Curse.Radium.Addons.AddonGameInstance"
},
{
  "timestamp": "2025-06-11T13:53:44.7814016-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "[AddonGameInstance] LoadInstancesFromDisk loaded 0 instances"
},
{
  "timestamp": "2025-06-11T13:53:44.7816557-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing game instance of type Curse.Games.Minecraft.MinecraftGameInstance"
},
{
  "timestamp": "2025-06-11T13:53:45.480211-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "[MinecraftGameInstance] LoadInstancesFromDisk loaded 2 instances"
},
{
  "timestamp": "2025-06-11T13:53:45.4930722-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game Minecraft (Id: 432) was detected at /home/avery/Documents/curseforge/minecraft/Instances/e/"
},
{
  "timestamp": "2025-06-11T13:53:45.5447348-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initialized game e instance"
},
{
  "timestamp": "2025-06-11T13:53:45.54488-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game Minecraft (Id: 432) was detected at /home/avery/Documents/curseforge/minecraft/Instances/Homestead - A Cozy Survival Experience/"
},
{
  "timestamp": "2025-06-11T13:53:45.5786275-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initialized game Homestead - A Cozy Survival Experience instance"
},
{
  "timestamp": "2025-06-11T13:53:45.5822493-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Game instances initialization completed. Operation took 930 ms"
},
{
  "timestamp": "2025-06-11T13:53:45.5834665-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Initializing Modpack instances."
},
{
  "timestamp": "2025-06-11T13:53:45.5895999-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-11T13:53:50.6143789-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #2 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-11T13:54:00.6348503-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Attempt #3 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-11T13:54:10.6580748-04:00",
  "level": "Error",
  "thread": 10,
  "memory": 0.0,
  "message": "UpdateCache Exception!",
  "error": {
    "message": "One or more errors occurred. (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443))",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
      "Curse.Radium.Addons.ProjectRepository+<UpdateProjectsAsync>d__22.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()",
      "System.Threading.TimerQueueTimer.Fire(Boolean)",
      "System.Threading.TimerQueue.FireNextTimers()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused (api.curseforge.com:443)",
      "type": "System.Net.Http.HttpRequestException",
      "stack": [
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Server.Communication.BaseWebServiceClient+<PostAsync>d__15`1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.TechService.AddonServiceClient+<GetProjectsAsync>d__11.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass22_0+<<UpdateProjectsAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.Radium.Utilities.Retry+<>c__DisplayClass4_0+<<DoAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()"
      ],
      "innerError": {
        "message": "Connection refused",
        "type": "System.Net.Sockets.SocketException",
        "stack": [
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
        ]
      }
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:10.6900867-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Running startup game detection"
},
{
  "timestamp": "2025-06-11T13:54:10.7019367-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Performing quick game detection for 25 games..."
},
{
  "timestamp": "2025-06-11T13:54:10.7190683-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Completed startup game detection. We have a total of 2 detected game instances. Operation took 28 ms"
},
{
  "timestamp": "2025-06-11T13:54:10.722944-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-11T13:54:10.7247798-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-11T13:54:11.1518219-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Setting file watcher on instance directory /home/avery/Documents/curseforge/minecraft/Instances"
},
{
  "timestamp": "2025-06-11T13:54:11.1659216-04:00",
  "level": "Warn",
  "thread": 6,
  "memory": 0.0,
  "message": "Service call failed to: v1/minecraft/modloader/timestamp",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:11.1682248-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting modloaders database timestamp from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-11T13:54:11.169018-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed getting modloaders database timestamp from server",
  "error": {
    "message": "Failed to get modloaders timestamp from server",
    "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
    "stack": [
      "Curse.TechService.AddonServiceClient+<GetModloadersDatabaseTimestampAsync>d__27.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Games.Minecraft.ModLoader.ModLoaderRepository+<GetServiceTimestampAsync>d__17.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.TechService.AddonServiceClient+<GetModloadersDatabaseTimestampAsync>d__27.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task`1.TrySetResult(TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1,TResult)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult(TResult)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ]
  }
},
{
  "timestamp": "2025-06-11T13:54:11.185456-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Refreshing Minecraft instances list"
},
{
  "timestamp": "2025-06-11T13:54:11.1868464-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Initializing tasks"
},
{
  "timestamp": "2025-06-11T13:54:11.1857737-04:00",
  "level": "Warn",
  "thread": 14,
  "memory": 0.0,
  "message": "Service call failed to: v1/minecraft/modloader?includeAll=true",
  "error": {
    "message": "Connection refused (api.curseforge.com:443)",
    "type": "System.Net.Http.HttpRequestException",
    "stack": [
      "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Threading.Tasks.ValueTask`1.get_Result()",
      "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "Curse.Server.Communication.ServicesConfiguration.LogWarning(Exception,String)",
      "Curse.Server.Communication.BaseWebServiceClient+<GetAsync>d__14`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Threading.Tasks.Task.TrySetException(Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncValueTaskMethodBuilder`1.SetException(Exception)",
      "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread,ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread)",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused",
      "type": "System.Net.Sockets.SocketException",
      "stack": [
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
        "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:11.1875571-04:00",
  "level": "Error",
  "thread": 14,
  "memory": 0.0,
  "message": "Failed getting Minecraft Modloaders from server. Request failed with status code InternalServerError"
},
{
  "timestamp": "2025-06-11T13:54:11.1885701-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Resolving Minecraft instances list from /home/avery/Documents/curseforge/minecraft/Instances"
},
{
  "timestamp": "2025-06-11T13:54:11.1929118-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Skipping existing minecraft game instance at /home/avery/Documents/curseforge/minecraft/Instances/e"
},
{
  "timestamp": "2025-06-11T13:54:11.1930047-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Skipping existing minecraft game instance at /home/avery/Documents/curseforge/minecraft/Instances/Homestead - A Cozy Survival Experience"
},
{
  "timestamp": "2025-06-11T13:54:11.1989735-04:00",
  "level": "Info",
  "thread": 11,
  "memory": 0.0,
  "message": "Attempt #1 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-11T13:54:11.2006203-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Discord RPC client initialized successfully."
},
{
  "timestamp": "2025-06-11T13:54:11.2011306-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Initializing Periodic Tasks"
},
{
  "timestamp": "2025-06-11T13:54:11.2539301-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-11T13:54:11.2539643-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-11T13:54:11.2500713-04:00",
  "level": "Error",
  "thread": 2,
  "memory": 0.0,
  "message": "Global Unhandled Exception",
  "data": {
    "Exception": {
      "Type": "System.AggregateException",
      "Message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Failed to get Minecraft Modloaders from server)",
      "Inner": {
        "Type": "Curse.Server.Communication.Exceptions.ServerRequestException",
        "Message": "Failed to get Minecraft Modloaders from server",
        "Stack": [
          "Curse.TechService.AddonServiceClient+<GetModloadersAsync>d__26.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
          "Curse.Games.Minecraft.ModLoader.ModLoaderRepository+<GetRemoteItemsAsync>d__21.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
          "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<UpdateCacheAsync>d__14.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
          "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<InitializeAsync>d__8.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
          "Curse.AppService.MinecraftAppService+<<InitializeAsync>b__22_0>d.MoveNext()"
        ]
      }
    }
  },
  "error": {
    "message": "A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Failed to get Minecraft Modloaders from server)",
    "type": "System.AggregateException",
    "innerError": {
      "message": "Failed to get Minecraft Modloaders from server",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient+<GetModloadersAsync>d__26.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Games.Minecraft.ModLoader.ModLoaderRepository+<GetRemoteItemsAsync>d__21.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<UpdateCacheAsync>d__14.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.Radium.Addons.BaseAddonServiceDatabase`3+<InitializeAsync>d__8.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.AppService.MinecraftAppService+<<InitializeAsync>b__22_0>d.MoveNext()"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:11.720324-04:00",
  "level": "Info",
  "thread": 19,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-11T13:54:11.7203573-04:00",
  "level": "Info",
  "thread": 19,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-11T13:54:13.6793592-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "Validating Minecraft installation"
},
{
  "timestamp": "2025-06-11T13:54:13.6793944-04:00",
  "level": "Info",
  "thread": 10,
  "memory": 0.0,
  "message": "/home/avery/Documents/curseforge/minecraft/Install/minecraft-launcher found."
},
{
  "timestamp": "2025-06-11T13:54:13.6919101-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:54:16.2147455-04:00",
  "level": "Info",
  "thread": 19,
  "memory": 0.0,
  "message": "Attempt #2 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-11T13:54:18.7453303-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:54:26.2304397-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Attempt #3 of 3 on InstalledAddonDatabase -> GetAddons"
},
{
  "timestamp": "2025-06-11T13:54:28.7599559-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.Tasks.Task+<>c.<.cctor>b__281_0(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:28.8087385-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:54:33.9019199-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:54:36.2482913-04:00",
  "level": "Error",
  "thread": 10,
  "memory": 0.0,
  "message": "UpdateCache Exception!",
  "error": {
    "message": "One or more errors occurred. (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443)) (Connection refused (api.curseforge.com:443))",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
      "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
      "Curse.Radium.Addons.ProjectRepository+<UpdateProjectsAsync>d__22.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task.FinishSlow(Boolean)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception,Task`1&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception)",
      "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.ExecutionContextCallback(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext(Thread)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1.MoveNext()",
      "System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox,Boolean)",
      "System.Threading.Tasks.Task.RunContinuations(Object)",
      "System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()",
      "System.Threading.TimerQueueTimer.Fire(Boolean)",
      "System.Threading.TimerQueue.FireNextTimers()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Connection refused (api.curseforge.com:443)",
      "type": "System.Net.Http.HttpRequestException",
      "stack": [
        "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<ConnectAsync>d__103.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<CreateHttp11ConnectionAsync>d__105.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<AddHttp11ConnectionAsync>d__79.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Threading.Tasks.TaskCompletionSourceWithCancellation`1+<WaitWithCancellationAsync>d__1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__89.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Threading.Tasks.ValueTask`1.get_Result()",
        "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult()",
        "System.Net.Http.DecompressionHandler+<SendAsync>d__16.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter.GetResult()",
        "System.Net.Http.HttpClient+<<SendAsync>g__Core|83_0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Server.Communication.BaseWebServiceClient+<PostAsync>d__15`1.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.TechService.AddonServiceClient+<GetProjectsAsync>d__11.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass22_0+<<UpdateProjectsAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter.GetResult()",
        "Curse.Radium.Utilities.Retry+<>c__DisplayClass4_0+<<DoAsync>b__0>d.MoveNext()",
        "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
        "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
        "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
        "System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()",
        "Curse.Radium.Utilities.Retry+<DoAsync>d__5`1.MoveNext()"
      ],
      "innerError": {
        "message": "Connection refused",
        "type": "System.Net.Sockets.SocketException",
        "stack": [
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.ThrowException(SocketError,CancellationToken)",
          "System.Net.Sockets.Socket+AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Sockets.Socket+<<ConnectAsync>g__WaitForConnectWithCancellation|285_0>d.MoveNext()",
          "System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()",
          "System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task)",
          "System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task,ConfigureAwaitOptions)",
          "System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()",
          "System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable+ConfiguredValueTaskAwaiter.GetResult()",
          "System.Net.Http.HttpConnectionPool+<ConnectToTcpHostAsync>d__104.MoveNext()"
        ]
      }
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:43.9159967-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Author()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.Tasks.Task+<>c.<.cctor>b__281_0(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:43.9181858-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:54:48.9351147-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:54:58.9485651-04:00",
  "level": "Error",
  "thread": 6,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateFileID()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_UpdateAvailable()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.Tasks.Task+<>c.<.cctor>b__281_0(Object)",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.TaskScheduler.TryExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:54:59.9827115-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:55:04.9995567-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:55:15.0163592-04:00",
  "level": "Error",
  "thread": 15,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:55:15.0196926-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:55:20.0304017-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:55:30.0438317-04:00",
  "level": "Error",
  "thread": 15,
  "memory": 0.0,
  "message": "Failed to get addon from service!",
  "error": {
    "message": "One or more errors occurred. (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError) (Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError)",
    "type": "System.AggregateException",
    "stack": [
      "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)",
      "Curse.Radium.Addons.ProjectRepository.FetchProjectWithRetries(Int32)",
      "Curse.Radium.Addons.ProjectRepository.UpdateProject(Int32)",
      "Curse.Radium.Addons.ProjectRepository.GetProjectById(Int32,Boolean)",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Addon()",
      "Curse.Games.Minecraft.MinecraftGameInstance.get_Author()",
      "Curse.AppService.Minecraft.MinecraftInstanceMapper.Map(MinecraftGameInstance)",
      "System.Linq.Enumerable+SelectEnumerableIterator`2.MoveNext()",
      "System.Linq.Enumerable+WhereEnumerableIterator`1.ToArray()",
      "Curse.Plugin.API.Handlers.Minecraft.GameInstance.MinecraftGetInstancesRequestHandler.HandleRequest(MinecraftGetInstancesRequest,CancellationToken)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0+<<Handle>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.API.Handlers.AsyncApiRequestHandlerBase`1+<>c__DisplayClass7_0.<Handle>b__0()",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0+<<ExecuteAsync>b__0>d.MoveNext()",
      "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine&)",
      "System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine&)",
      "Curse.Plugin.RequestExecutor+<>c__DisplayClass9_0.<ExecuteAsync>b__0()",
      "System.Threading.Tasks.Task`1.InnerInvoke()",
      "System.Threading.ExecutionContext.RunInternal(ExecutionContext,ContextCallback,Object)",
      "System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&,Thread)",
      "System.Threading.Tasks.Task.ExecuteEntry()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+ConcurrentExclusiveTaskScheduler.ExecuteTask(Task)",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair.ProcessExclusiveTasks()",
      "System.Threading.Tasks.ConcurrentExclusiveSchedulerPair+SchedulerWorkItem.System.Threading.IThreadPoolWorkItem.Execute()",
      "System.Threading.ThreadPoolWorkQueue.Dispatch()",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerDoWork(PortableThreadPool,Boolean&)",
      "System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()",
      "System.Threading.Thread+StartHelper.RunWorker()",
      "System.Threading.Thread+StartHelper.Run()",
      "System.Threading.Thread.StartCallback()"
    ],
    "innerError": {
      "message": "Failed getting project 1258033. Status Code: InternalServerError. Exception: System.Net.WebException: Web Service responded with failure status code: InternalServerError",
      "type": "Curse.Server.Communication.Exceptions.ServerRequestException",
      "stack": [
        "Curse.TechService.AddonServiceClient.HandleResponse[T](ServiceResponse`1,Func`1)",
        "Curse.TechService.AddonServiceClient.GetProject(Int32)",
        "Curse.Radium.Addons.ProjectRepository+<>c__DisplayClass31_0.<FetchProjectWithRetries>b__0()",
        "Curse.Radium.Utilities.Retry.Do[T](Func`1,IRetryStrategy,String,Func`2)"
      ]
    }
  }
},
{
  "timestamp": "2025-06-11T13:55:30.0448029-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #1 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:55:35.0793704-04:00",
  "level": "Info",
  "thread": 15,
  "memory": 0.0,
  "message": "Attempt #2 of 2 on InstalledAddonDatabase -> FetchAddonWithRetries"
},
{
  "timestamp": "2025-06-11T13:55:37.6925208-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
},
{
  "timestamp": "2025-06-11T13:55:37.6981871-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network connected and internet available"
},
{
  "timestamp": "2025-06-11T13:55:37.9183011-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network address has changed. Checking connection"
},
{
  "timestamp": "2025-06-11T13:55:37.9193421-04:00",
  "level": "Info",
  "thread": 14,
  "memory": 0.0,
  "message": "Network status changed: Disconnected"
},
{
  "timestamp": "2025-06-11T13:55:38.0012145-04:00",
  "level": "Info",
  "thread": 6,
  "memory": 0.0,
  "message": "Network availability has changed. Checking connection"
}
]}
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                